pykrige/rk.py

```diff
@@ -64,28 +136,37 @@ def fit(self, x, y, *args, **kwargs):
         y: ndarray
             array of targets (N, )
         """
-
-        points = self._dimensionality_check(x)
-
-        # if condition required to address backward compatibility
-        if self.method in threed_krige:
-            self.model = krige_methods[self.method](
-                val=y,
-                variogram_model=self.variogram_model,
-                nlags=self.nlags,
-                weight=self.weight,
-                verbose=self.verbose,
-                **points
-            )
-        else:
-            self.model = krige_methods[self.method](
-                z=y,
-                variogram_model=self.variogram_model,
-                nlags=self.nlags,
-                weight=self.weight,
-                verbose=self.verbose,
-                **points
-            )
+        setup = dict(
+            variogram_model=self.variogram_model,
+            variogram_parameters=self.variogram_parameters,
+            variogram_function=self.variogram_function,
+            nlags=self.nlags,
+            weight=self.weight,
+            verbose=self.verbose,
+        )
+        add_setup = dict(
+            anisotropy_scaling=self.anisotropy_scaling,
+            anisotropy_angle=self.anisotropy_angle,
+            enable_statistics=self.enable_statistics,
+            coordinates_type=self.coordinates_type,
+            anisotropy_scaling_y=self.anisotropy_scaling_y,
+            anisotropy_scaling_z=self.anisotropy_scaling_z,
+            anisotropy_angle_x=self.anisotropy_angle_x,
+            anisotropy_angle_y=self.anisotropy_angle_y,
+            anisotropy_angle_z=self.anisotropy_angle_z,
+            drift_terms=self.drift_terms,
+            point_drift=self.point_drift,
+            external_drift=self.external_drift,
+            external_drift_x=self.external_drift_x,
+            external_drift_y=self.external_drift_y,
+            functional_drift=self.functional_drift,
+        )
```

Comment:
At this point maybe,
```
add_setup = {key: getattr(self, key) for key in ["anisotropy_scaling", ...]}
```
but writing is explicitly as done above is also fine..

Reply:
Following the Zen of Python: Explicit is better than Implicit ;-)